tests/client/test_auth.py

```diff
@@ -127,3 +203,136 @@ def test_auth_invalid_type():
     with Client(dispatch=MockDispatch(), auth="not a tuple, not a callable") as client:
         with pytest.raises(TypeError):
             client.get(url)
+
+
+def test_digest_auth_returns_no_auth_if_no_digest_header_in_response():
+    url = "https://example.org/"
+    auth = DigestAuth(username="tomchristie", password="password123")
+
+    with Client(dispatch=MockDispatch()) as client:
+        response = client.get(url, auth=auth)
+
+    assert response.status_code == 200
+    assert response.json() == {"auth": None}
```

Comment:
Can we add a test case where only 1 of the 2 conditions are met like 200+WWW-Authenticate or 401 without the header?